syntax = "proto3";
package metadata.v1;
option java_package = "com.movieaugur.service.metadata.v1";

message MetadataRequest {
  repeated string ids = 1;
}

message ItemMetadata {
  string id = 1;
  string title = 2;
  string description = 3;
  string objectType = 4;
  repeated string categories = 5;
  repeated string creators = 6;
}

message MetadataResponse {
  repeated ItemMetadata metadata = 1;
}

service Metadata {
  rpc GetMetadata (MetadataRequest) returns (MetadataResponse);
}
